apps/src/netsim/NetSimLocalClientNode.js

```diff
@@ -494,4 +520,102 @@ NetSimLocalClientNode.prototype.handleMessage_ = function (message) {
   if (this.receivedLog_) {
     this.receivedLog_.log(message.payload);
   }
-};
\ No newline at end of file
+};
+
+/**
+ * Asynchronously receive the latest message shared between this node
+ * and its connected remote node.
+ * @param {!NodeStyleCallback} onComplete - given the message as a result, or
+ *        NULL if no messages exist.
+ */
+NetSimLocalClientNode.prototype.getLatestMessageOnSimplexWire = function (onComplete) {
+  if (!this.myWire) {
+    onComplete(new Error("Unable to retrieve message; not connected."));
+    return;
+  }
+
+  // Does an asynchronous request to the message table to ensure we have
+  // the latest contents
+  this.shard_.messageTable.readAll(function (err, messageRows) {
+    if (err) {
+      onComplete(err);
+      return;
+    }
+
+    // We only care about rows on our (simplex) wire
+    var rowsOnWire = messageRows.filter(function (row) {
+      return this.myWire.isMessageRowOnSimplexWire(row);
+    }.bind(this));
+
+    // If there are no rows, complete successfully but pass null result.
+    if (rowsOnWire.length === 0) {
+      onComplete(null, null);
+      return;
+    }
+
+    var lastRow = rowsOnWire[rowsOnWire.length -1];
+    onComplete(null, new NetSimMessage(this.shard_, lastRow));
+  }.bind(this));
+};
+
+/**
+ * Asynchronously set the state of the shared wire.
+ * @param {string} newState - probably ought to be "0" or "1"
+ * @param {!NodeStyleCallback} onComplete
+ */
+NetSimLocalClientNode.prototype.setSimplexWireState = function (newState, onComplete) {
+  this.sendMessage(newState, function (err) {
+    if (err) {
+      err.message = "Failed to set wire state: " + err.message;
```

Comment:
This feels a little weird to me, though I can't really justify exactly why. I wonder if it would be cleaner to create a new Error with this message instead of modifying the existing one.